augur/curate/rename.py

```diff
@@ -0,0 +1,85 @@
+"""
+Renames fields / columns of the input data
+"""
+
+from typing import Iterable, Literal, Union, List, Tuple
+import argparse
+from augur.io.print import print_err
+from augur.errors import AugurError
+
+def register_parser(parent_subparsers):
+    parser = parent_subparsers.add_parser("rename",
+    parents = [parent_subparsers.shared_parser],
+    help = __doc__)
+
+    p = parser.add_argument_group(title="RENAME SPECIFIC OPTIONS")
```

Comment:
is this a consistent pattern across `augur`, or new for this command?

(i ask because i've done a couple of these, both with command-specific options, and didn't use this pattern; if this is an intentional pattern across the code base, i should go back and fix those other two commands.)

Reply:
Hmm, `augur curate format-dates` does have a split of [REQUIRED](https://github.com/nextstrain/augur/blob/4264701ab98975edb46a100a241415b705095914/augur/curate/format_dates.py#L30) vs [OPTIONAL](https://github.com/nextstrain/augur/blob/4264701ab98975edb46a100a241415b705095914/augur/curate/format_dates.py#L40) argument groups for command-specific options.

Reply:
Yeah, so does [curate titlecase](https://github.com/nextstrain/augur/blob/c679fded55cc8792a22e43a4cc5589604aabbb43/augur/curate/titlecase.py#L16-L30), and I used them for inspiration here.

The aim of this PR isn't to rock the boat so I'll happily use the approach @joverlee521 recommends!

But beyond this PR, if we run `augur curate titlecase --help` we get

```console
usage: augur curate titlecase [-h] ...

Applies titlecase to string fields in a metadata record

options:
  -h, --help            show this help message and exit

INPUTS: 
  Input options ... 

OUTPUTS:
  Output options ...

REQUIRED:
  --titlecase-fields ...

OPTIONAL:
  --articles [ARTICLES ...] ...
```

however I think it'd be much nicer to surface the titlecase-specific options first.

Reply:
> however I think it'd be much nicer to surface the titlecase-specific options first.

Yeah that does seem nicer, will have to look into it. From [brief look on StackOverflow](https://stackoverflow.com/a/39048205), seems like we would need to change the order of `parser._action_groups`

Reply:
Cool! Would be a nice little improvement down the line. 

For the purpose of this PR, taking into account [this comment](https://github.com/nextstrain/augur/pull/1506#discussion_r1662986157) we'll have one optional and one required argument. @joverlee521 should I have REQUIRED/OPTIONAL or RENAME OPTIONS or ???

Reply:
Let's just standardize on REQUIRED/OPTIONAL

Reply:
> Let's just standardize on REQUIRED/OPTIONAL

Done in force-push. Thanks for making a decision here!